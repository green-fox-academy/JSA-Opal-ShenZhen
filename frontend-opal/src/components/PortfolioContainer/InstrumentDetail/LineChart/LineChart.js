import React from 'react';
import { View, Text } from 'react-native';
import {
  VictoryLine,
  VictoryChart,
  VictoryAxis,
  VictoryScatter,
  VictoryBrushContainer,
  VictoryZoomContainer
} from 'victory-native';

import styles from '../styles';

let InstrumentDetail = () => {
  const dataInstrument = [
    { ts: 'May', value: 2 },
    { ts: 'June', value: 4 },
    { ts: 'July', value: 6.5 },
    { ts: 'Aug', value: 6 },
    { ts: 'Sep', value: 5 },
    { ts: 'Oct', value: 7 },
    { ts: 'Nov', value: 9 }
  ];

  // return (
  //   <View>

  return (
    <View>
      <VictoryChart
        padding={{ top: 0, left: 50, right: 50, bottom: 30 }}
        height={200}
        scale={{ x: 'time' }}
        containerComponent={
          <VictoryBrushContainer
            responsive={false}
            brushDimension="x"
            // brushDomain={this.state.selectedDomain}
            // onBrushDomainChange={this.handleBrush.bind(this)}
            brushStyle={{ fill: 'teal', opacity: 0.1 }}
          />
        }
      >
        <VictoryLine
          style={{
            data: { stroke: 'tomato' }
          }}
          data={[
            { x: new Date(1982, 4, 1), y: 125 },
            { x: new Date(1987, 5, 1), y: 257 },
            { x: new Date(1993, 6, 1), y: 345 },
            { x: new Date(1997, 7, 1), y: 515 },
            { x: new Date(2001, 8, 1), y: 132 },
            { x: new Date(2005, 9, 1), y: 305 },
            { x: new Date(2011, 10, 1), y: 270 },
            { x: new Date(2015, 11, 1), y: 470 }
          ]}
        />

        <VictoryAxis
          tickValues={[
            new Date(1985, 4, 1),
            // new Date(1990, 5, 1),
            new Date(1995, 6, 1),
            // new Date(2000, 7, 1),
            new Date(2005, 8, 1),
            // new Date(2010, 9, 1)
          ]}
          tickFormat={x => new Date(x).getFullYear()}
        />
        <VictoryAxis style={{ axis: { stroke: 'none' } }} dependentAxis />
      </VictoryChart>
    </View>
  );
};

{
  /* <VictoryChart
        containerComponent={
          <VictoryBrushContainer
            // brushDomain={{ x: [-2, 7.5], y: [-1, 8] }}
            // brushDimension="y"
            brushStyle={{ fill: 'teal', opacity: 0.1 }}
          />
        }
        responsive={false}
      >
        <VictoryLine
          with={100}
          height={100}
          data={dataInstrument}
          style={{ data: { stroke: 'green' } }}
          y={data => data.value}
          responsive={false}
        /> */
}
{
  /* <VictoryScatter
          data={dataInstrument}
          y={data => data.value}
          size={3}
          style={{ data: { fill: 'green' } }}
        /> */
}
{
  /* <VictoryAxis
          tickValues={dataInstrument.map(item => item.ts)}
          style={{ axis: { stroke: 'none' } }}
        />
        <VictoryAxis style={{ axis: { stroke: 'none' } }} dependentAxis /> */
}
{
  /* </VictoryChart> */
}
{
  /* </View> */
}
// );
// };

export default InstrumentDetail;
