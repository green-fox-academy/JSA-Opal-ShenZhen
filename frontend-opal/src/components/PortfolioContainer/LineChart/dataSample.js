const dataSample = [
  { x: new Date(2019, 5, 1), y: 3 },
  { x: new Date(2019, 6, 1), y: 5 },
  { x: new Date(2019, 7, 1), y: 4 },
  { x: new Date(2019, 8, 1), y: 6 },
  { x: new Date(2019, 9, 1), y: 6.5 },
  { x: new Date(2019, 10, 1), y: 5.5 },
  { x: new Date(2019, 11, 1), y: 8 }
];

export default dataSample;
