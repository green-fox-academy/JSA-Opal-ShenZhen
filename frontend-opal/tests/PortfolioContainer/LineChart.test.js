import React from 'react';
import renderer from 'react-test-renderer';
import LineChart from 'components/PortfolioContainer/LineChart';

describe('LineChart component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LineChart />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
