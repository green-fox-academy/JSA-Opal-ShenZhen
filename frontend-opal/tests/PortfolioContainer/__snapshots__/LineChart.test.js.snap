// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart component renders correctly 1`] = `
<View>
  <View
    className="VictoryContainer"
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={180}
      bbWidth={750}
      height={180}
      meetOrSlice={0}
      minX={0}
      minY={0}
      role="img"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 180,
            "width": 750,
          },
        ]
      }
      vbHeight={180}
      vbWidth={750}
      width={750}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4278190080,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGGroup
          clipPath="victory-clip-2"
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGDefs>
              <RNSVGClipPath
                name="victory-clip-2"
              >
                <RNSVGRect
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  height={140}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  rx={0}
                  ry={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={1}
                  width={0}
                  x={60}
                  y={10}
                />
              </RNSVGClipPath>
            </RNSVGDefs>
          </RNSVGGroup>
          <RNSVGPath
            d={null}
            fill={
              Array [
                0,
                0,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                4294927175,
              ]
            }
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={2}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGLine
            fill={
              Array [
                0,
                0,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={1}
            x={0}
            x1={60}
            x2={700}
            y={0}
            y1={150}
            y2={150}
          />
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGText
              content={null}
              dx={
                Array [
                  0,
                ]
              }
              dy={
                Array [
                  11.969999999999999,
                ]
              }
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              name="chart-axis-2-tickLabels-0"
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={
                Array [
                  60,
                ]
              }
              y={
                Array [
                  161,
                ]
              }
            >
              <RNSVGTSpan
                content="Thu Jan 01 1970 08:00:00 GMT+0800 (GMT+08:00)"
                dx={
                  Array [
                    0,
                  ]
                }
                dy={Array []}
                fill={
                  Array [
                    0,
                    4280624421,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": 14,
                    "letterSpacing": "normal",
                    "textAnchor": "middle",
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={
                  Array [
                    60,
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGLine
            fill={
              Array [
                0,
                0,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={1}
            x={0}
            x1={60}
            x2={60}
            y={0}
            y1={10}
            y2={150}
          />
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGText
              content={null}
              dx={
                Array [
                  0,
                ]
              }
              dy={
                Array [
                  4.97,
                ]
              }
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              name="chart-axis-3-tickLabels-0"
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={
                Array [
                  49,
                ]
              }
              y={
                Array [
                  122,
                ]
              }
            >
              <RNSVGTSpan
                content="0.2"
                dx={
                  Array [
                    0,
                  ]
                }
                dy={Array []}
                fill={
                  Array [
                    0,
                    4280624421,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": 14,
                    "letterSpacing": "normal",
                    "textAnchor": "end",
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={
                  Array [
                    49,
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGText
              content={null}
              dx={
                Array [
                  0,
                ]
              }
              dy={
                Array [
                  4.97,
                ]
              }
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              name="chart-axis-3-tickLabels-1"
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={
                Array [
                  49,
                ]
              }
              y={
                Array [
                  94,
                ]
              }
            >
              <RNSVGTSpan
                content="0.4"
                dx={
                  Array [
                    0,
                  ]
                }
                dy={Array []}
                fill={
                  Array [
                    0,
                    4280624421,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": 14,
                    "letterSpacing": "normal",
                    "textAnchor": "end",
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={
                  Array [
                    49,
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGText
              content={null}
              dx={
                Array [
                  0,
                ]
              }
              dy={
                Array [
                  4.97,
                ]
              }
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              name="chart-axis-3-tickLabels-2"
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={
                Array [
                  49,
                ]
              }
              y={
                Array [
                  66,
                ]
              }
            >
              <RNSVGTSpan
                content="0.6"
                dx={
                  Array [
                    0,
                  ]
                }
                dy={Array []}
                fill={
                  Array [
                    0,
                    4280624421,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": 14,
                    "letterSpacing": "normal",
                    "textAnchor": "end",
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={
                  Array [
                    49,
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGText
              content={null}
              dx={
                Array [
                  0,
                ]
              }
              dy={
                Array [
                  4.97,
                ]
              }
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              name="chart-axis-3-tickLabels-3"
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={
                Array [
                  49,
                ]
              }
              y={
                Array [
                  38,
                ]
              }
            >
              <RNSVGTSpan
                content="0.8"
                dx={
                  Array [
                    0,
                  ]
                }
                dy={Array []}
                fill={
                  Array [
                    0,
                    4280624421,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": 14,
                    "letterSpacing": "normal",
                    "textAnchor": "end",
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={
                  Array [
                    49,
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGText
              content={null}
              dx={
                Array [
                  0,
                ]
              }
              dy={
                Array [
                  4.97,
                ]
              }
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              name="chart-axis-3-tickLabels-4"
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={
                Array [
                  49,
                ]
              }
              y={
                Array [
                  10,
                ]
              }
            >
              <RNSVGTSpan
                content="1.0"
                dx={
                  Array [
                    0,
                  ]
                }
                dy={Array []}
                fill={
                  Array [
                    0,
                    4280624421,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": 14,
                    "letterSpacing": "normal",
                    "textAnchor": "end",
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={
                  Array [
                    49,
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
          "zIndex": 99,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={180}
        bbWidth={750}
        height={180}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "height": "100%",
              "overflow": "visible",
              "width": "100%",
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 180,
              "width": 750,
            },
          ]
        }
        vbHeight={180}
        vbWidth={750}
        width={750}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
